src/parser.h

```diff
@@ -384,6 +393,17 @@ class parser_t : public std::enable_shared_from_this<parser_t> {
     statuses_t get_last_statuses() const { return vars().get_last_statuses(); }
     void set_last_statuses(statuses_t s) { vars().set_last_statuses(std::move(s)); }
 
+    /// Get a parser status variable
+    const wcstring& get_status_var(parser_status_var_t var) const {
+        return status_vars_.at((int) var);
+    }
+
+    /// Set a parser status variable
+    void set_status_var(parser_status_var_t var, wcstring val) {
+        ASSERT_IS_MAIN_THREAD();
```

Comment:
@ridiculousfish do you want me to remove this assert?

Reply:
yeah